agent/internal/task_config.go

```diff
@@ -28,6 +28,7 @@ type TaskConfig struct {
 	Task               *task.Task
 	BuildVariant       *model.BuildVariant
 	Expansions         *util.Expansions
+	DynamicExpansions  *util.Expansions
```

Comment:
If you want this to be extra safe, having this be a `util.Expansions` without the pointer is better since you'll only have to deal with one level of nilness rather than two - for example, a nil `*util.Expansion` vs. a non-pointer `util.Expansions` that happens to be default initialize (i.e. a nil map).

Reply:
Doesn't setting it in NewTaskConfig take care of this concern?